src/NuGet.Core/NuGet.Protocol.VisualStudio/VSSourceRepositoryProvider.cs

```diff
@@ -86,7 +88,7 @@ public Configuration.IPackageSourceProvider PackageSourceProvider
 
         private void Init()
         {
-            _repositories.Clear();
+            _repositories = new List<SourceRepository>();
```

Comment:
would you make this a local variable and assign it to _respositories at the end?  Or return the repositories here and assign them above.

Not related to this change, but it looks like this isn't thread safe and it could end up with duplicates.